Jenkinsfile: remove recursive submodules

mtusnio opened the pull request:
This closes #82

Ham22 commented on Jenkinsfile:
```diff
@@ -44,14 +44,14 @@ node('docker && imgtec') {
                     [$class: 'CleanCheckout'],
                     [$class: 'PruneStaleBranch'],
-                    [$class: 'SubmoduleOption', recursiveSubmodules: true]
+                    [$class: 'SubmoduleOption', recursiveSubmodules: false]
```
Just delete the line instead

Ham22 commented:
Can you add a reason to the commit message

mtusnio committed:
Jenkinsfile: allow multithreaded builds

Signed-off-by: Michal Tusnio <michal.tusnio@imgtec.com>

mtusnio committed:
Jenkinsfile: do not override doGenerateSubmoduleConfiguration option for checkout.

This flag significantly increases the checkout time, however there is not need to override it explicitly.

Signed-off-by: Michal Tusnio <michal.tusnio@imgtec.com>

mtusnio committed:
Jenkinsfile: remove recursive submodules checkout

None of the Contiki submodules specify their own submodules, there is no need to check them out recursively.

Signed-off-by: Michal Tusnio <michal.tusnio@imgtec.com>
